### PR TITLE
feature: Implement hasOne relation (issue #6)

### PR DESCRIPTION
**Pull request by qmohitsingh**
**Implement hasOne relation for issue #6**

- Added logic to handle 'hasOne' in the relation processing loop.

### CODE REVIEW

**Comment by qmohitsingh** [comment]
Please review this PR  and pls let me know if something needs to be changed @nicoabie

File: src/assemble.test.ts [hunk 1]
```diff
@@ -68,3 +68,96 @@ test('assemble hasMany null', () => {
...
+  const identifiers = ['p.personId', 'c.contactId'];
+
+  assert.deepStrictEqual(assemble(relations, aliases, identifiers, objects), [
+    { personId: 1 },
+    { personId: 2 },
+    { personId: 3 },
+    { personId: 4, contactDetails: { contactId: 103 } },
+    { personId: 5 },
+  ]);
+});
```
Comment (nicoabie) [inline src/assemble.test.ts:163]: I don't think this is correct.

1. the database will return something for person 5, see my previous comment
2. person 1 and 3 point to the same object and that is fine from a strict point of view

File: src/assemble.ts [hunk 2]
```diff
@@ -77,6 +78,19 @@ export const assemble: AssembleFn = <T>(
           if (child[childIdentifier.split('.')[1]] !== null) {
             parent[parentChildRelation[1]].push(child);
           }
+        } else if (parentChildRelation[0] === 'hasOne') {
+          if (child[childIdentifier.split('.')[1]] !== null) {
```
Comment (nicoabie) [inline src/assemble.ts:82]: maybe we should put `childIdentifier.split('.')[1]` in a variable as it is being calculated 4 times

**Review by nicoabie — CHANGES_REQUESTED** [review]
thanks again for contributing and sorry for the time it took me to review.
are you allowed to request my reviews?

File: src/assemble.test.ts (hunk 1 above)
Comment (qmohitsingh) [reply src/assemble.test.ts]: fixed

File: src/assemble.ts (hunk 2 above)
Comment (qmohitsingh) [reply src/assemble.ts]: fixed

File: src/assemble.test.ts [hunk 3]
```diff
@@ -68,3 +68,96 @@ test('assemble hasMany null', () => {
...
+    p: 'person',
+    c: 'contact',
+  };
+  const identifiers = ['p.personId', 'c.contactId'];
+
+  assert.deepStrictEqual(assemble(relations, aliases, identifiers, objects), [
+    { personId: 1 },
+    { personId: 2 },
+  ]);
+});
```
[the remark this answers is not in the document]
Comment (qmohitsingh) [reply src/assemble.test.ts]: test deleted

**Comment by qmohitsingh** [comment]
> thanks again for contributing and sorry for the time it took me to review. are you allowed to request my reviews?

ofcoure!! no worries, and no i don't think i have permission to request the review

File: src/assemble.test.ts [hunk 4]
```diff
@@ -68,3 +68,75 @@ test('assemble hasMany null', () => {
...
+  };
+  const aliases = {
+    p: 'person',
+    c: 'contact',
+  };
+  const identifiers = ['p.personId', 'c.contactId'];
+
+  assert.deepStrictEqual(assemble(relations, aliases, identifiers, objects), [
+    { personId: 1, contactDetails: { contactId: 101 } },
+    { personId: 2 },
```
Comment (nicoabie) [inline src/assemble.test.ts:114]: in this case contactDetails should be null 

{ personId: 2, contactDetails: null },

File: src/assemble.test.ts [hunk 5]
```diff
@@ -68,3 +68,75 @@ test('assemble hasMany null', () => {
...
+  };
+  const identifiers = ['p.personId', 'c.contactId'];
+
+  assert.deepStrictEqual(assemble(relations, aliases, identifiers, objects), [
+    { personId: 1 },
+    { personId: 2 },
+    { personId: 3 },
+    { personId: 4, contactDetails: { contactId: 103 } },
+  ]);
+});
```
Comment (nicoabie) [inline src/assemble.test.ts:142]: we can remove this one

File: src/assemble.test.ts (hunk 4 above)
Comment (qmohitsingh) [reply src/assemble.test.ts]: fixed this

File: src/assemble.test.ts (hunk 5 above)
Comment (qmohitsingh) [reply src/assemble.test.ts]: test removed

**Review by nicoabie — APPROVED** [review]

**Pull request merged**